src/main/objects/bass/BassTranslator.js

```diff
@@ -532,69 +800,198 @@ function BassTranslator() {
+                                    argsMap.extra.push(toAlter + "<")
+                                    if (Utils.contains(argsMap.omit, toAlter + "")) argsMap.omit.splice(argsMap.extra.indexOf(toAlter), 1)
+                                }
+                                addedSomething = true
+                                notAdd = true
+                            }
+                        } else {
+                            if (Utils.contains(argsMap.extra, toAlter + "" + "<")) {
+                                if (argsMap.revolution !== toAlter + "" + "<") {
+                                    argsMap.extra.splice(argsMap.extra.indexOf(toAlter + "" + "<"), 1)
+                                    argsMap.extra.push(toAlter + "")
+                                    if (Utils.contains(argsMap.omit, toAlter + "")) argsMap.omit.splice(argsMap.extra.indexOf(toAlter), 1)
+                                }
+                                addedSomething = true
+                                notAdd = true
+                            } else if (Utils.contains(argsMap.extra, toAlter + "")) {
+                                if (argsMap.revolution !== toAlter + "") {
+                                    argsMap.extra.splice(argsMap.extra.indexOf(toAlter + ""), 1)
+                                    argsMap.extra.push(toAlter + ">")
+                                    if (Utils.contains(argsMap.omit, toAlter + "")) argsMap.omit.splice(argsMap.extra.indexOf(toAlter), 1)
+                                }
+                                addedSomething = true
+                                notAdd = true
+                            }
+                        }
                     }
-                    if (harmonicFunctions[0][i].extra === undefined) {
-                        harmonicFunctions[0][i].extra = []
+                    if (!notAdd) {
+                        argsMap.extra.push(toExtra[a] + "")
+                        addedSomething = true
                     }
-
-                    harmonicFunctions[0][i].omit.push(toOmit[a])
-                    harmonicFunctions[0][i].extra.push(toExtra[a])
                 }
             }
 
+            if (addedSomething) {
+                this.handleDownChord(argsMap)
+                this.fixExtraAfterModeChange(argsMap)
+                this.addOmit3ForS2IfNecessary(argsMap)
+                harmonicFunctions[0][i] = new HarmonicFunction.HarmonicFunction2(argsMap)
+                if (DEBUG) Utils.log("harm function after copy adding toExtra and toOmit",
+                    JSON.stringify(harmonicFunctions[0][i]))
+            }
+
         }
     }
 
-    this.isD7 = function(harmonicFunction) {
+    this.isD7 = function (harmonicFunction) {
         return harmonicFunction.functionName === Consts.FUNCTION_NAMES.DOMINANT &&
             (Utils.contains(harmonicFunction.extra, "7") || Utils.contains(harmonicFunction.extra, ">7") ||
```

Comment:
Można użyć Utils.containsBaseComponent - i unikamy x or y or z

Reply:
Wychodzi na to, że ta funkcja i inne nie działały wcale i nic nie robiły... Jak je chyba naprawiłem, to zaczęły się testy wywalać, więc je zupełnie wytnę